### ISSUE TITLE
PlatformException on MacOS through amplify_secure_storage and amplify_auth

### ISSUE THREAD

**Issue by pixnbit** [aws-amplify/amplify-flutter#5361]
### Description

Hi, we've been using Amplify Auth in production, and recently upgraded to the latest. No issues with Android, iOS, and Web, but on the MacOS the app would not launch with the following exception:

```
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: PlatformException(channel-error, Unable to establish connection on channel., null, null)
package:amplify_secure_storage/src/pigeons/ns_user_defaults_pigeon.g.dart 56:7                   NSUserDefaultsPigeon.boolFor
===== asynchronous gap ===========================
package:amplify_secure_storage/src/amplify_secure_storage.vm.dart 156:29                         AmplifySecureStorage._initializeScope
===== asynchronous gap ===========================
package:amplify_secure_storage/src/amplify_secure_storage.vm.dart 96:13                          AmplifySecureStorage._init.<fn>
===== asynchronous gap ===========================
package:amplify_secure_storage/src/amplify_secure_storage.vm.dart 70:5                           AmplifySecureStorage._init
===== asynchronous gap ===========================
package:amplify_secure_storage/src/amplify_secure_storage.vm.dart 111:5                          AmplifySecureStorage.read
===== asynchronous gap ===========================
package:amplify_auth_cognito_dart/src/state/machines/credential_store_state_machine.dart 84:21   CredentialStoreStateMachine.getVersion
===== asynchronous gap ===========================
package:amplify_auth_cognito_dart/src/state/machines/credential_store_state_machine.dart 320:21  CredentialStoreStateMachine._migrateLegacyCredentialStore
===== asynchronous gap ===========================
package:amplify_auth_cognito_dart/src/state/machines/credential_store_state_machine.dart 416:5   CredentialStoreStateMachine.onLoadCredentialStore
===== asynchronous gap ===========================
package:amplify_auth_cognito_dart/src/state/machines/credential_store_state_machine.dart 60:11   CredentialStoreStateMachine.resolve
===== asynchronous gap ===========================
package:amplify_core/src/state_machine/state_machine.dart 262:11                                 StateMachine._listenForEvents.<fn>
===== asynchronous gap ===========================
package:amplify_core/src/state_machine/state_machine.dart 251:7                                  StateMachine._listenForEvents
===== asynchronous gap ===========================
package:amplify_core/src/state_machine/event.dart 39:47                                          new EventCompleter
package:amplify_core/src/state_machine/state_machine.dart 186:19                                 StateMachineManager.dispatch
package:amplify_core/src/state_machine/state_machine.dart 32:23                                  Dispatcher.dispatchAndComplete
package:amplify_core/src/state_machine/state_machine.dart 197:13                                 StateMachineManager.dispatchAndComplete
package:amplify_auth_cognito_dart/src/state/cognito_state_machine.dart 77:36                     CognitoAuthStateMachine.loadCredentials
package:amplify_auth_cognito_dart/src/state/machines/configuration_state_machine.dart 108:25     ConfigurationStateMachine.onConfigure
package:amplify_auth_cognito_dart/src/state/machines/configuration_state_machine.dart 50:15      ConfigurationStateMachine.resolve
package:amplify_core/src/state_machine/state_machine.dart 262:59                                 StateMachine._listenForEvents.<fn>.<fn>
dart:async/future.dart 431:42                                                                    new Future.delayed.<fn>
dart:async-patch/timer_patch.dart 18:15                                                          Timer._createTimer.<fn>
dart:isolate-patch/timer_impl.dart 398:19                                                        _Timer._runTimers
dart:isolate-patch/timer_impl.dart 429:5                                                         _Timer._handleMessage
dart:isolate-patch/isolate_patch.dart 184:12                                                     _RawReceivePort._handleMessage
===== asynchronous gap ===========================
package:amplify_core/src/state_machine/event.dart 39:47                                          new EventCompleter
package:amplify_core/src/state_machine/state_machine.dart 157:23                                 StateMachineManager.accept
package:amplify_core/src/state_machine/state_machine.dart 169:23                                 StateMachineManager.acceptAndComplete
package:amplify_auth_cognito_dart/src/auth_plugin_impl.dart 247:25                               AmplifyAuthCognitoDart.configure
package:amplify_auth_cognito/src/auth_plugin_impl.dart 88:5                                      AmplifyAuthCognito.configure
dart:async/future.dart 534:21                                                                    Future.wait.<fn>
package:amplify_core/src/amplify_class.dart 199:7                                                AmplifyClass._configurePlugins
package:amplify_core/src/amplify_class.dart 118:7                                                AmplifyClass.configure
```

I have the following amplify versions:

```
  amplify_auth_cognito: 2.4.0
  amplify_authenticator: 2.1.2
  amplify_flutter: 2.4.0
```

I searched around didn't find anywhere people mentioning it. I read through the [migration guide](https://docs.amplify.aws/gen1/flutter/start/project-setup/upgrade-guide/) and did not find anything related to this

Is it a bug in the framework? Thanks for looking into it 

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [X] Auth
- [ ] Authenticator
- [ ] DataStore
- [ ] Notifications (Push)
- [X] Storage

### Steps to Reproduce

Exception thrown on app launch at:
```
  await Amplify.addPlugin(AmplifyAuthCognito());
  await Amplify.configure(amplifyconfig);
```

### Platforms

- [ ] iOS
- [ ] Android
- [ ] Web
- [X] macOS
- [ ] Windows
- [ ] Linux

### Flutter Version

3.24.1

### Amplify Flutter Version

2.4.0

### Deployment Method

Amplify CLI

### Schema

```GraphQL
NA
```

**Comment by Jordan-Nelson** [comment]
Hello @pixnbit - I will see if we can reproduce this. I have a few questions to help with the reproduction:
- What version of MacOS have you observed this on?
- What version of Amplify Flutter are you upgrading from?
- Are you also upgrading your version of Flutter at the same time? If so, from what version?
- Do you observe that this issue goes away if you downgrade Amplify packages to the previous version you were using?

**Comment by MarlonJD** [comment]
Same issue, cannot login on macOS, but on iOS it works

**Comment by MarlonJD** [comment]
I just added PR, I tested on macOS it works, if I don't know if it's intended but macOS pluginClass was missing on amplify_secure_storage's pubspec, I think we need urgent bug fix release for this.

**Comment by MarlonJD** [comment]
I just added PR, I tested on macOS it works, I don't know if it's intended but macOS pluginClass was missing on amplify_secure_storage's pubspec, I think we need urgent bug fix release for this.

**Comment by Jordan-Nelson** [comment]
@MarlonJD Thanks for the PR. I was able to reproduce this. We will review and look to get a release out as soon as possible.

**Comment by Jordan-Nelson** [comment]
This has been released in amplify_secure_storage v0.5.6. You can run `flutter pub upgrade` to get the latest version. Thank you for your contribution @MarlonJD!

**Comment by pixnbit** [comment]
Thanks for the speedy response on this!
